src/gwc/src/main/java/org/geoserver/gwc/controller/GwcDemoUrlHandlerMapping.java

```diff
@@ -0,0 +1,16 @@
+/* (c) 2022 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.gwc.controller;
+
+import org.geoserver.catalog.Catalog;
+
+/** Handler for mapping workspace-based demo requests to non-workspace-based requests. */
+public class GwcDemoUrlHandlerMapping extends GwcWmtsRestUrlHandlerMapping {
```

Comment:
Since demo is basically a simple GUI, would it make sense to move this dispatcher to the web-gwc module?
My hope is that if someone removes all the GUI modules (common in some high security environments) then the GWC GUI is gone too.

Reply:
todo;
1. move GwcDemoUrlHandlerMapping.java to web/gwc
2. move the bean definition to here as well

Reply:
moved